src/StatsdClient/Telemetry.cs

```diff
@@ -142,15 +146,19 @@ public void Dispose()
 
         private void SendMetric(string metricName, int value)
         {
-            if (_optionalTransport != null)
+            if (_optionalTransport != null && _optionalMetricSerializer != null)
```

Comment:
Is there a functional scenario where we expect these to be null? 
If not - could we move this check into the constructor and fail there? 

Because right now `SendMetric` will just fail silently if these are `null` which seems like a weird behavior.

Reply:
Telemetry can be disabled. This is a variation of null object pattern https://en.wikipedia.org/wiki/Null_object_pattern.